lib/vault/encode.rb

```diff
@@ -0,0 +1,11 @@
+module Vault
+  module EncodePath
+    def encode_path(path)
```

Comment:
Could we add a detailed comment here explaining:

1. What this does
1. Why it's important

(I know the answers to both of these questions, but I'm thinking about future engineers or community contributors. Bonus points for YARD.)